compiler/rustc_hir_analysis/src/variance/constraints.rs

```diff
@@ -253,8 +253,8 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
                 self.add_constraints_from_ty(current, typ, variance);
             }
 
-            ty::RawPtr(ref mt) => {
-                self.add_constraints_from_mt(current, mt, variance);
+            ty::RawPtr(ty, mutbl) => {
+                self.add_constraints_from_mt(current, &ty::TypeAndMut { ty, mutbl }, variance);
```

Comment:
okay... not the only use of `TypeAndMut` 🤷 idk if it's useful to fully remove it then

Reply:
Is there an issue tracking "entirely remove `TypeAndMut`"? That would be the logical next step IMO.

Reply:
I am not totally sure, some of the remaining uses here seem appropriate as an abstraction over both `Ref` and `RawPtr`. Personally don't care too much about keeping it around 🤷